addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -128,6 +274,161 @@ def _record_production(self):
                 'date_start': datetime.now(),
             })
 
+    def _update_finished_move(self):
+        if not self.serial_batch_creation:
+            super()._update_finished_move()
+        else:
+            production_move = self.production_id.move_finished_ids.filtered(
+                lambda move: move.product_id == self.product_id and
+                move.state not in ('done', 'cancel')
+            )
+            location_dest_id = production_move.location_dest_id._get_putaway_strategy(self.product_id).id or production_move.location_dest_id.id
+            for line in self.finished_workorder_line_ids.filtered(lambda l: l.product_id == self.product_id):
+                if not line.lot_name:
+                    raise UserError(_('You need to provide a serial number for the finished product.'))
+                line.lot_id = self.finished_lot_id.create({
+                    'name': line.lot_name,
+                    'product_id': self.product_id.id,
+                    'company_id': self.company_id.id,
+                })
+                production_move.move_line_ids.create({
+                    'move_id': production_move.id,
+                    'product_id': production_move.product_id.id,
+                    'lot_id': line.lot_id.id,
+                    'product_uom_qty': 1.0,
+                    'product_uom_id': self.product_uom_id.id,
+                    'qty_done': 1.0,
+                    'location_id': production_move.location_id.id,
+                    'location_dest_id': location_dest_id,
+                })
+
+    def _generate_serial_numbers(self):
```

Comment:
maybe the lot name generation on a prefix/suffix should be in stock and reused in both mrp and stock module